### PR TITLE
Address nits from per-utxo change

### PR DESCRIPTION
**Pull request by morcos**
@sipa the other nits from my review

### CODE REVIEW

**Comment by sipa** [comment]
utACK 567a8fb86878d69619925fb1b945d6613ce134ad

**Review by gmaxwell — APPROVED** [review]
utACK

**Comment by morcos** [comment]
Removed one of the comment changes to reflect new semantics of HaveCoinInCache from #10559

File: src/coins.h [hunk 1]
```diff
@@ -149,8 +149,7 @@ class CCoinsView
     //! Retrieve the Coin (unspent transaction output) for a given outpoint.
     virtual bool GetCoin(const COutPoint &outpoint, Coin &coin) const;
 
-    //! Just check whether we have data for a given outpoint.
-    //! This may (but cannot always) return true for spent outputs.
+    //! Just check whether we have data for a given outpoint and that it is not spent.
```
Comment (sipa) [inline src/coins.h]: As pointed out in another PR, `CTxMemPool::HaveCoin` would violate this.

**Comment by laanwj** [comment]
Needs rebase

**Comment by morcos** [comment]
rebased again...  this PR is slowly getting smaller and smaller :)

File: src/coins.h [hunk 2]
```diff
@@ -151,7 +151,11 @@ class CCoinsView
      */
     virtual bool GetCoin(const COutPoint &outpoint, Coin &coin) const;
 
+<<<<<<< HEAD
     //! Just check whether a given outpoint is unspent.
+=======
+    //! Just check whether we have data for a given outpoint and that it is not spent.
+>>>>>>> d3cbbf1... Comment clarifications in coins.cpp
```
Comment (sipa) [inline src/coins.h]: This does not look like valid C++.

Comment (morcos) [inline src/coins.h]: sigh....

**Comment by morcos** [comment]
fixed broken push

**Comment by sipa** [comment]
utACK 21d4afa12fbf1e7f59b629060c9e10db213fe07a

**Comment by laanwj** [comment]
utACK 21d4afa

**Comment by a415263** [comment]
fffgee

**Pull request merged**